random-beans/src/main/java/io/github/benas/randombeans/util/ReflectionUtils.java

```diff
@@ -165,6 +166,24 @@ public static boolean isParameterizedType(final Type type) {
     }
 
     /**
+     * Check if a type is an unbounded wildcard type
+     * @param parameterizedType the ParameterizedType to check
+     * @return true if the ParameterizedType is an unbounded wild card type, false otherwise
+     */
+    public static boolean isUnboundedWildcardType(final ParameterizedType parameterizedType) {
```

Comment:
I think the `isWildcardType` method is good enough. It should identify all wildcards e.g. `<?>` `<? extends Runnable>`  The `isUnboundedWildcardType` checks if the wildcard is `<?>`. 

You solution is more generally useful, because it allows random-beans to generate empty collections for all collections with wildcard generics.

My solution is more limited, when I was writing it I was thinking to much about adding support for collections with bound wildcards to extends the classpath scanning.

I will reduce the pull request to the test this evening.